tchannel-core/src/main/java/com/uber/tchannel/utils/TChannelUtilities.java

```diff
@@ -38,26 +39,57 @@
 
     public static final ByteBuf emptyByteBuf = Unpooled.EMPTY_BUFFER;
 
+    public static int scoreAddr(NetworkInterface iface, InetAddress addr) {
+        int score = 0;
+
+        if (addr instanceof Inet4Address) {
+            score += 300;
+        }
+
+        try {
+            if (!iface.isLoopback() && !addr.isLoopbackAddress()) {
+                score += 100;
+                if (iface.isUp()) {
+                    score += 100;
+                }
+            }
+        } catch (SocketException e) {
+            logger.error("unable to score IP.", e);
```

Comment:
Nit: Include more information

    logger.error("Unable to score IP {} of interface {}", addr, iface, e)